barretenberg/cpp/src/barretenberg/ultra_honk/decider_prover.cpp

```diff
@@ -50,16 +49,20 @@ template <IsUltraFlavor Flavor> void DeciderProver_<Flavor>::execute_relation_ch
  * */
 template <IsUltraFlavor Flavor> void DeciderProver_<Flavor>::execute_pcs_rounds()
 {
+    if (proving_key->proving_key.commitment_key == nullptr) {
```

Comment:
I think this will be a nullptr in the UltraHonk case, but not for ClientIVC because we hold a commitment key in the accumulator.

Reply:
please add a comment - I think it would be nice to have some test that validate that the commitment key is initialised at a certain stage and has been deinitialised at another, but not sure if that's easy